src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -532,6 +544,142 @@ public static int GetCurrentProcessorId()
             return currentProcessorIdCache >> ProcessorIdCacheShift;
         }
 
+        // do a fast check and record in a readonly static so that it could become a JIT constant
+        internal static readonly bool IsCoreIdReallyFast = SimpleCoreIdSpeedCheck();
+
+        // If GetCurrentProcessorNumber takes any nontrivial time (compared to TLS access), return false.
+        // Check more than once - to make sure it was not because TLS was delayed by GC or a context switch.
+        private static bool SimpleCoreIdSpeedCheck()
+        {
+            // warm up the code paths.
+            int id = UninlinedThreadStatic() | GetCurrentProcessorNumber();
+            long _05usec = Stopwatch.Frequency / 1000000;
```

Comment:
This would be problematic on a system where `Stopwatch.Frequency` is less then `1_000_000`. I think most systems that have HPET or later timer will be at least 10MHz, however.

Reply:
I originally had a check that frequency is high enough, then observed that low frequency would lead to calibration anyways and decided to not optimize for that case as nowdays rare, - just let it naturally fail and calibrate.

---
In reply to: [352948987](https://github.com/dotnet/runtime/pull/467#discussion_r352948987) [](ancestors = 352948987)

Reply:
Thanks!

A small comment just stating that it was considered and should resolve correctly might be useful.

Reply:
Will add a comment. It gave me a pause too when you mentioned it, but then I remembered.

---
In reply to: [353305989](https://github.com/dotnet/runtime/pull/467#discussion_r353305989) [](ancestors = 353305989)